scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_model.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import (absolute_import, division, print_function)
+
+from Muon.GUI.Common.muon_data_context import MuonDataContext, construct_empty_group
+
+
+class GroupingTableModel(object):
+    """
+    Model used when grouping table is not part of the grouping tab. When the
+    grouping table presenter is embedded in the grouping tab, use the GroupingTabModel instead.
+
+    The model takes an instance of the MuonContext as an input, most of its functionality is simply
+    a callback to this class.
+    """
+
+    def __init__(self, data=MuonDataContext()):
+        self._data = data
+
+    @property
+    def groups(self):
+        return self._data.groups
+
+    @property
+    def group_names(self):
+        return list(self._data.group_names)
+
+    @property
+    def group_and_pair_names(self):
```

Comment:
this seems out of place here as everything else only handles groups

Reply:
The grouping table needs to know the pair names to ensure that it does not allow a group to be added with the same name as a pair.